### DISCUSSION TITLE
Start playing of a playlist with large number of files is extremely slow

### DISCUSSION THREAD

**Discussion by de666** [music-assistant/support#2573] [Feature requests and ideas]
### What version of Music Assistant has the issue?

2.1.0b6

### What version of the Home Assistant Integration have you got installed?

2024.6.2

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

When a playlist with many songs inside (let's say 500 or many) is played (filesystem provider), it takes minutes to start playing.
Also when trying to play such playlist from home assistant browse media interface is not possible because it stucks with the loading symbol.
The only one way to start playing (delayed by minutes as said before) is from the MA web UI without going inside the playlist.

### How to reproduce

Click play on a playlist which has a couple of hunders of file (500/1000)

### Music Providers

File system (local disk)

### Player Providers

Chromecast

### Additional information

I don't know how it works but I assume it should work like this:

- MA read the playlist
- for every line/row/path it searches for the file inside the library
- since it has all the tag information in the library database, it has just to find the right row and take the song info from there

also if this can be slow, I think it should start playing the first song and then keep searching/loading the next song info and so on but It seems it has to fetch all the songs information before start playing the first song and again I can't figure out why this operation shuold be so slow (hope it has not to fetch all the covers before start playing...).

### What version of Home Assistant Core are your running

2024.6.3

### What type of installation are you running?

Home Assistant Container

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
Speed with large playlists does depend on the resources available on the host that the server is running on

**Comment by de666** [comment]
> Speed with large playlists does depend on the resources available on the host that the server is running on

I know it but I don't understand:

1. Why it is so slow on a "standard" mini pc with 4 GB of ram
2. Why this operation requires so much resources, in my mind is not something CPU intensive
3. why MA has to fetch all the songs info before start playing and this action can't be done asyncrously while playing is yet started

**Comment by de666** [comment]
Just an update:
I noticed that the huge playlist is loaded almost instantly after the first play action (which takes minutes before playing) like if there was a sort of cache or whatever which has to be "populated" the first time a playlist is loaded/played.
Also after the first play, the same playlist is browsable almost flawlessly from the home assistant media browser.
Is it possible to do this sort of "loading" task immediately (and in background) when the new playlist is seen/discovered by the library scanner?
This behaviour confirms my idea that this is not stritcly related to the hardware resources but more likely to how the playlists are managed.

**Comment by marcelveldt** [comment]
This is by design - the playlists needs to be loaded once before its cached. 
One day we might optimize this but its not as simple as you may think and low priority

**Comment by OzGav** [comment]
Moved to feature request

**Comment by piellick** [comment]
Hi Everyone, 
same feedback from spotify provider, i have my "liked song playlist" with something like 10 000 songs and it's hard to load. 

A workaround is to use "500 random tracks from libraby" whith a correct loading.

**Comment by OzGav** [comment]
I am closing this. MA will continue to be optimised however there are hardware resource limitations that will have a greater impact on responsiveness than the software. The docs advise to keep the queue to one thousand tracks or less. If you have a a powerful host then you will be able to go higher than this recommendation.
